### ISSUE TITLE
Regression line not drawing correctly

### ISSUE THREAD

**Issue by rodyager** [amcharts/amcharts4#2751]
Linear Regression lines with reorder: true and simplify: true only draws between the first valueX and last valueX, rather than between min(valueX) and max(valueX). 

eg.

```
series: [ 
   type: "LineSeries"
  dataFields: {
           valueX: "x",
           valueY:  "y"
   },
   name: "Linear Regression",
   plugins: [ {
         type: "Regression",
         reorder: true,
         simplify: true
        }],
 data : [ {x: 10, y:10}, {x:0, y:0}, {x:100, y:100}, {x:20, y:20} ]
```

will draw a line from (10,10) to (20,20) and not from (0,0) to (100,100).

While removing either reorder or simplify provides the full trend line, neither is a good solution.

**Comment by martynasma** [comment]
Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by rodyager** [comment]
Here’s a minimal working example. The trend line it draws is completely wrong.

If you replace the “data” line with the data in the correct order, the regression line is drawn almost correctly. (It still doesn’t quite go through the points).

Rod

<head>
<script src="https://www.amcharts.com/lib/4/core.js"></script>
<script src="https://www.amcharts.com/lib/4/charts.js"></script>
<script src="//www.amcharts.com/lib/4/plugins/regression.js<http://www.amcharts.com/lib/4/plugins/regression.js>"></script>

<title>Bug Example</title>
</head>

<body>

<div id="plot" style="width: 90%; height: 400px;" ></div>

<script>
var chart = am4core.createFromConfig({
xAxes: [ { type: "ValueAxis" } ],
yAxes: [ { type: "ValueAxis" } ],
series: [
{
type: "LineSeries",
bullets: [ { children: [ { type: "Circle", radius: 3, strokeOpacity: 0 } ] } ],
dataFields: { valueX: "x", valueY: "y" },
strokeOpacity: 0
},
{
type: "LineSeries",
dataFields: { valueX: "x",  valueY: "y" },
strokeWidth: 1,
plugins: [ { type: "Regression", reorder: true, simplify: true } ]
}
],
data: [ {x:40, y:40}, {x:20, y:20}, {x:100, y:100}, {x:80, y:80}]
// data: [ {x:20, y:20}, {x:40, y:40}, {x:80, y:80}, {x:100, y:100}]
}, "plot", "XYChart");
</script>
</body>
On 29 Jul 2020, at 5:32 pm, Martynas Majeris <notifications@github.com<mailto:notifications@github.com>> wrote:



Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by rodyager** [comment]
Here’s a minimal working example. The trend line it draws is completely wrong.

If you replace the “data” line with the data in the correct order, the regression line is drawn almost correctly. (It still doesn’t quite go through the points).

Rod

```
<head>
<script src="https://www.amcharts.com/lib/4/core.js"></script>
<script src="https://www.amcharts.com/lib/4/charts.js"></script>
<script src="//www.amcharts.com/lib/4/plugins/regression.js<http://www.amcharts.com/lib/4/plugins/regression.js>"></script>

<title>Bug Example</title>
</head>

<body>

<div id="plot" style="width: 90%; height: 400px;" ></div>

<script>
var chart = am4core.createFromConfig({
xAxes: [ { type: "ValueAxis" } ],
yAxes: [ { type: "ValueAxis" } ],
series: [
{
type: "LineSeries",
bullets: [ { children: [ { type: "Circle", radius: 3, strokeOpacity: 0 } ] } ],
dataFields: { valueX: "x", valueY: "y" },
strokeOpacity: 0
},
{
type: "LineSeries",
dataFields: { valueX: "x",  valueY: "y" },
strokeWidth: 1,
plugins: [ { type: "Regression", reorder: true, simplify: true } ]
}
],
data: [ {x:40, y:40}, {x:20, y:20}, {x:100, y:100}, {x:80, y:80}]
// data: [ {x:20, y:20}, {x:40, y:40}, {x:80, y:80}, {x:100, y:100}]
}, "plot", "XYChart");
</script>
</body>
```

**Comment by martynasma** [comment]
There's an issue: `reorder` parameter is not really working as it is supposed to.

This will be fixed in the next version, at which point I'll notify you here.

Thank you for bringing this to our attention.

**Comment by martynasma** [comment]
Fixed in [4.9.34](https://github.com/amcharts/amcharts4/releases/tag/4.9.34).

## [4.9.34] - 2020-07-31

### Added
- New bullet `ShapeBullet` in `Bullets` plugin. Please refer to [this documentation article](https://www.amcharts.com/docs/v4/tutorials/plugin-bullets/) for more info.
- `treatZeroAs` added to `ValueAxis`. Use it to work around zero-value restriction on a logarithmic scale.
- `stackToNegative` (default: `true`) added to `XYSeries`. Determines whether negative values in a stack will stack in relation to base (zero) line (`true`; default) or in relation to the previous value in the stack (`false`).

### Changed
- [xlsx](https://www.npmjs.com/package/xlsx) dependency was updated from `0.15.4` to `0.16.4`.
- Tooltip snapping on `DateAxis` now takes Series' location into account.

### Fixed
- Regression plugin: `reorder` option was not working properly.
- `snapToSeries` was not working properly if there was a hidden Series with exactly the same value as the other visible Series.
- `TreeMap` and `ForceDirectedTree` with a single data item and no children were not working properly with chart legend.
- Horizontal (X) Axis tooltip in a very small document was sometimes positioned with an offset.
- `zoomToRectangle()` method of `MapChart` with `deltaLongitude` was not working properly if a rectangle crossed the 180 meridian.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
